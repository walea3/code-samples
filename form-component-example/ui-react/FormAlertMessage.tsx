import cx from 'classnames';
import React from 'react';

import {
  Message,
  MessageProps,
  MessageVariants,
  ProminenceVariants,
} from '@nel-ui/message-react';

import FormStyles from './FormStyles';

export interface FormAlertMessageProps extends MessageProps {
  /**
   * Alert style
   * @default error
   */
  variant: MessageVariants.ERROR | MessageVariants.WARNING;
}

export const FormAlertMessage = ({
  inlineFirstChild = false,
  prominence = ProminenceVariants.HIGH,
  variant = MessageVariants.ERROR,
  ...props
}: FormAlertMessageProps) => (
  <FormStyles>
    {({ classes }) => (
      <Message
        role="alert"
        {...props}
        className={cx(classes.formAlertMessage, classes[`${variant}FormAlertMessage`], {
          [classes.prominent]: prominence === ProminenceVariants.HIGH,
        })}
        inlineFirstChild={inlineFirstChild}
        prominence={prominence}
        variant={variant}
      />
    )}
  </FormStyles>
);

FormAlertMessage.displayName = 'FormAlertMessage';
