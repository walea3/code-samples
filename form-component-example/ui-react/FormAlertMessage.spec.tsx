import React from 'react';

import { renderWithNEL } from '@nel-ui/tooling-react/test-utils';

import { FormAlertMessage } from './FormAlertMessage';

describe('<FormAlertMessage />', () => {
  const requiredProps = {
    heading: 'Foo',
  };

  const renderComponent = (props) =>
    renderWithNEL(<FormAlertMessage {...requiredProps} {...props} />);

  describe('@renders', () => {
    test.each`
      name                             | props
      ${'as default'}                  | ${{}}
      ${'as inlineFirstChild'}         | ${{ inlineFirstChild: true }}
      ${'as warning'}                  | ${{ variant: 'warning' }}
      ${'as inlineFirstChild warning'} | ${{ inlineFirstChild: true, variant: 'warning' }}
    `('$name', ({ props }) => {
      expect(renderComponent(props)).toMatchSnapshot();
    });
  });
});
