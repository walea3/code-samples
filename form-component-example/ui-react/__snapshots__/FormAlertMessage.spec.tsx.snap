// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormAlertMessage /> @renders as default 1`] = `
<div
  class="nel-formAlertMessage nel-errorFormAlertMessage nel-prominent nel-message nel-errorMessage nel-prominent"
  data-ref="message"
  role="alert"
>
  <svg
    aria-hidden="true"
    class="nel-messageIcon nel-icon nel-info-24 nel-infoIconSet nel-error"
    data-ref="icon"
    focusable="false"
    height="1"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 1 1"
    width="1"
  />
  <div
    class="nel-messageContent"
    data-ref="messageContent"
  >
    <p
      class="nel-messageHeading nel-paragraph"
      data-ref="messageHeader.text"
    >
      <strong
        class="nel-strong nel-inlineText"
        data-ref="text"
      >
        Foo
      </strong>
    </p>
  </div>
</div>
`;

exports[`<FormAlertMessage /> @renders as inlineFirstChild 1`] = `
<div
  class="nel-formAlertMessage nel-errorFormAlertMessage nel-prominent nel-message nel-errorMessage nel-prominent"
  data-ref="message"
  role="alert"
>
  <svg
    aria-hidden="true"
    class="nel-messageIcon nel-icon nel-info-24 nel-infoIconSet nel-error"
    data-ref="icon"
    focusable="false"
    height="1"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 1 1"
    width="1"
  />
  <div
    class="nel-messageContent nel-inlineFirstChild"
    data-ref="messageContent"
  >
    <p
      class="nel-messageHeading nel-paragraph"
      data-ref="messageHeader.text"
    >
      <strong
        class="nel-strong nel-inlineText"
        data-ref="text"
      >
        Foo
      </strong>
    </p>
  </div>
</div>
`;

exports[`<FormAlertMessage /> @renders as inlineFirstChild warning 1`] = `
<div
  class="nel-formAlertMessage nel-warningFormAlertMessage nel-prominent nel-message nel-warningMessage nel-prominent"
  data-ref="message"
  role="alert"
>
  <svg
    aria-hidden="true"
    class="nel-messageIcon nel-icon nel-info-24 nel-infoIconSet nel-warning"
    data-ref="icon"
    focusable="false"
    height="1"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 1 1"
    width="1"
  />
  <div
    class="nel-messageContent nel-inlineFirstChild"
    data-ref="messageContent"
  >
    <p
      class="nel-messageHeading nel-paragraph"
      data-ref="messageHeader.text"
    >
      <strong
        class="nel-strong nel-inlineText"
        data-ref="text"
      >
        Foo
      </strong>
    </p>
  </div>
</div>
`;

exports[`<FormAlertMessage /> @renders as warning 1`] = `
<div
  class="nel-formAlertMessage nel-warningFormAlertMessage nel-prominent nel-message nel-warningMessage nel-prominent"
  data-ref="message"
  role="alert"
>
  <svg
    aria-hidden="true"
    class="nel-messageIcon nel-icon nel-info-24 nel-infoIconSet nel-warning"
    data-ref="icon"
    focusable="false"
    height="1"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 1 1"
    width="1"
  />
  <div
    class="nel-messageContent"
    data-ref="messageContent"
  >
    <p
      class="nel-messageHeading nel-paragraph"
      data-ref="messageHeader.text"
    >
      <strong
        class="nel-strong nel-inlineText"
        data-ref="text"
      >
        Foo
      </strong>
    </p>
  </div>
</div>
`;
